service-base/src/main/java/org/eclipse/hono/service/management/device/AbstractDeviceManagementHttpEndpoint.java

```diff
@@ -0,0 +1,242 @@
+ * Copyright (c) 2019 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+
+package org.eclipse.hono.service.management.device;
+
+import java.net.HttpURLConnection;
+import java.util.EnumSet;
+import java.util.Optional;
+import org.eclipse.hono.client.ClientErrorException;
+import org.eclipse.hono.config.ServiceConfigProperties;
+import org.eclipse.hono.service.http.AbstractHttpEndpoint;
+import org.eclipse.hono.service.http.HttpUtils;
+import org.eclipse.hono.service.http.TracingHandler;
+import org.eclipse.hono.service.management.Id;
+import org.eclipse.hono.service.management.OperationResult;
+import org.eclipse.hono.service.management.Result;
+import org.eclipse.hono.service.management.Util;
+import org.eclipse.hono.util.RegistryManagementConstants;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import io.opentracing.Span;
+import io.vertx.core.Promise;
+import io.vertx.core.Vertx;
+import io.vertx.core.http.HttpHeaders;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.core.http.HttpServerResponse;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.Router;
+import io.vertx.ext.web.RoutingContext;
+
+/**
+ * An {@code HttpEndpoint} for managing device registration information.
+ * <p>
+ * This endpoint implements Hono's <a href="https://www.eclipse.org/hono/docs/api/device-registration/">Device
+ * Registration API</a>. It receives HTTP requests representing operation invocations and sends them to an address on
+ * the vertx event bus for processing. The outcome is then returned to the peer in the HTTP response.
+ */
+public abstract class AbstractDeviceManagementHttpEndpoint extends AbstractHttpEndpoint<ServiceConfigProperties> {
+
+    private static final String SPAN_NAME_CREATE_DEVICE = "create Device from management API";
+    private static final String SPAN_NAME_GET_DEVICE = "get Device from management API";
+    private static final String SPAN_NAME_UPDATE_DEVICE = "update Device from management API";
+    private static final String SPAN_NAME_REMOVE_DEVICE = "remove Device from management API";
+
+    /**
+     * Creates an endpoint for a Vertx instance.
+     *
+     * @param vertx The Vertx instance to use.
+     * @throws NullPointerException if vertx is {@code null};
+     */
+    @Autowired
+    public AbstractDeviceManagementHttpEndpoint(final Vertx vertx) {
+        super(vertx);
+    }
+
+    @Override
+    protected String getEventBusAddress() {
+        return "";
+    }
+
+    @Override
+    public String getName() {
+        return String.format("%s/%s",
```

Comment:
IMHO this should be defined as a constant. No need to recreate it on each invocation, is there?